added check for undefined in isPromotable

Chaitje opened the pull request:
When calling api.promoTransaction with a bundle hash instead of a transaction hash the function isPromotable does not throw an error. It tries to read to ready isConsistent.state and throws "can not read state from undefined". This causes my app to crash and probably others as well.

I have now made sure that it simply returns false, causing the promotion to stop and return control to the app. It might be better to change the function that actually checks consistency, or add a check to differentiate between bundleHash and transactionHash (if that is possible?). But for now it fixes this issue.

Any feedback is welcome, as this is my first PR here on github. :)

senikk commented:
1) Its because res() is always called even then there is an error called with rej(). Could be solved with a return statement or else like this:
`            
if (err) {
  rej(err);
} else {
  res(isConsistent.state);  
}
` /cc @chrisdukakis 

2) The three last lines in api.prototype.isPromotable makes no sense at all. Promise is made for chaining and then() always returns a promise and since it just returning the same value as it get its just waste of code. The lines should just be removed and return the new promise directly. /cc @paulhandy

3) I wonder why api.prototype.isPromotable is promise based then the rest of the API is callback based.

Chaitje commented:
Allright thanks for the input.

From what I see it never returns a promise, and also shouldn't. It's just a check that will return a boolean. So it seems to me that using the if else construction (#1) should be a clean solution. I'll update that.

senikk commented:
:)

The (4) is just a check which returns a boolean, correct, but the last lines of the method is not returning a boolean its returning a pending promise. (just do a console.log() and see what you get) Promise.then() always just returns a promise (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then) so when using iota.api.isPromotable() we now have to check first if its returning a promise or a boolean not very clean.

To return a boolean value also for the rest of the method we need either to use async/await or as the rest of the methods in this API using callback function instead.

Chaitje commented:
Ah yes, you are right. Ok next try :):

I modified isPromotable to always return a promise. I also removed the isHash check in promoteTransaction because is was redundant. 

Now if an error occurs it gets propagated and a caller can choose what to do with it. etter like this I guess. Let me know, thx!

senikk commented:
"var command =" should be intented but except from that this looks much better :)

Chaitje commented:
Thx! Updated.

chrisdukakis commented on lib/api/api.js:
```diff
@@ -643,15 +643,8 @@ api.prototype.sendTransfer = function(seed, depth, minWeightMagnitude, transfers
-    }
 
     self.isPromotable(tail).then(function (isPromotable) {
-      if (!isPromotable) {
```
This is required for stopping promotion, if tail is inconsistent. Shouldn't be removed.
